docs/user-guide/command-line-interface.md

```diff
@@ -122,18 +122,22 @@ Examples:
 
 #### `--ext`
 
-This option allows you to specify which file extensions ESLint will use when searching for JavaScript files. By default, it uses `.js` as the only file extension.
+This option allows you to specify which file extensions ESLint will use when searching for JavaScript files in the directories you specify.
+By default, it uses `.js` as the only file extension.
 
 Examples:
 
     # Use only .js2 extension
-    eslint --ext .js2
+    eslint . --ext .js2
 
     # Use both .js and .js2
-    eslint --ext .js --ext .js2
+    eslint . --ext .js --ext .js2
 
     # Also use both .js and .js2
-    eslint --ext .js,.js2
+    eslint . --ext .js,.js2
+
+**Note:** This option has no effect on glob parameters.
+For example, `eslint lib/* --ext .js` will match all files within the `lib/` directory, regardless of extension.
```

Comment:
I think we're in agreement about the intent, but the devil is in the
details. Personally I find the language proposed in the PR to be
potentially confusing since you're showing a happy path. I think it would
be clearer to say that a command with just `--ext` still needs a
(something).

I also find "glob" confusing, as someone new to globs. To me, I lint files
or directories. But I respect that it might make the documentation simpler
if we always just say "globs" and have a single definition somewhere
prominent.
On Mar 20, 2016 8:01 PM, "Ian VanSchooten" <notifications@github.com> wrote:

> In docs/user-guide/command-line-interface.md
> <https://github.com/eslint/eslint/pull/5625#discussion_r56776700>:
>
> >
> >      # Also use both .js and .js2
> > -    eslint --ext .js,.js2
> > +    eslint . --ext .js,.js2
> > +
> > +**Note:** This option has no effect on glob parameters.
> > +For example, `eslint lib/* --ext .js` will match all files within the `lib/` directory, regardless of extension.
>
> That's a different meaning than I'm trying to convey, though. The issue is
> that --ext only applies when you give eslint a directory or set of
> directories. I think the solution to people not understanding globs would
> be to clarify what a glob is up at the top of the page, near
> https://github.com/eslint/eslint/pull/5625/files/5a8e8b84391eaca739e9d37c3aa986069b4bc2fb#diff-c6dea0f2431c0edc4b5356ce0fad3289R19
> perhaps. What do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/pull/5625/files/5a8e8b84391eaca739e9d37c3aa986069b4bc2fb#r56776700>
>